app/views/apps/_app.html.erb

```diff
@@ -1,16 +1,43 @@
-<div class="col-sm-3 col-md-3">
-  <%= link_to app_path(app.name, app.type, app.owner), class: "thumbnail app", target: "_blank", data: {
-    toggle: "popover",
-    content: manifest_markdown(app.manifest.description),
-    html: true,
-    trigger: "hover",
-    title: app.title,
-    container: "body"
-  } do %>
-    <div class="center-block" style="text-align: center">
-      <%= app_icon_tag(app) %>
-      <%= content_tag(:p, app.title) %>
-      <%= content_tag(:p, content_tag(:span, app.caption), class: "text-muted", style: "margin-top: 20px") if app.caption %>
+<%- if app.batch_connect_app? -%>
+  <%- app.batch_connect.sub_app_list.select(&:valid?).each do |sub_app| -%>
+    <div class="col-sm-3 col-md-3">
+      <%=
+        link_to(
+          new_batch_connect_session_context_path(token: sub_app.token),
```

Comment:
So difference between these two is this in psuedocode:

```ruby
if app.batch_connect_app?
  app.batch_connect.sub_app_list.select(&:valid?).each do |sub_app|
    render_icon_view_for sub_app with sub_app.url
  end
else
  render_icon_view_for app with app.url
end
```

Two options to simplify:

1. create a partial for the actual icon and render that, retaining the logic here
2. add a new method `OodApp#valid_sub_apps` and for default returns `[self]` then you have one loop, using `#url` for the url

Since we have separate OodApp and BatchConnect::App objects methods might need to exist on both

Reply:
Doing a quick review of the methods called in those blocks, I would need:

- `BatchConnect::App#batch_connect_app?` (to detect whether to open in new tab)
- `BatchConnect::App#caption`
- `BatchConnect::App#icon_path`
- `BatchConnect::App#name`
- `BatchConnect::App#type`
- `BatchConnect::App#owner`
- `BatchConnect::App#manifest`
- `OodApp#description`

For each of the `BatchConnect::App` methods I would need to treat appropriately for sub-apps as well. So exposing `BatchConnect::App#manifest` may be ill-advised since the following may occur:

```ruby
my_sub_app.description == my_sub_app.manifest.description
# => false
```

So if I don't implement the method `BatchConnect::App#manifest` for safety reasons, then we need to add:

- `OodApp#description`
- `OodApp#icon`
- `BatchConnect::App#icon`

Reply:
I think another option is to use the `OodApp#links` method and build up the `OodApp::Link` object with more properties.

Reply:
Reviewed the code changes and they look great